### ISSUE TITLE
当前版本的GUI界面，战利品模块不显示

### ISSUE THREAD

**Issue by breezeyyy** [ninthDevilHAUNSTER/ArknightsAutoHelper#245]
如题

**Comment by Redamancy-spec** [comment]
信息补充：
系统：Windows10 20H2
模拟器：Bluestacks 5.2.110.1003 N64
分辨率：1920x1080
软件版本：master+62be546-ci-118
详细报错信息：
Traceback (most recent call last):
  File "Arknights\helper.py", line 461, in on_end_operation
  File "imgreco\end_operation.py", line 206, in recognize
  File "imgreco\end_operation.py", line 288, in recognize_main
  File "imgreco\end_operation.py", line 72, in tell_group
  File "imgreco\item.py", line 170, in tell_item
  File "imgreco\item.py", line 85, in get_item_id
  File "imgreco\item.py", line 25, in _load_net
  File "imgreco\item.py", line 34, in load_index_info
  File "imgreco\item.py", line 63, in update_net
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\akhelper\\imgreco\\index_itemid_relation.json'

**Comment by Konano** [comment]
Solved in #256

**Comment by AutismYE** [comment]
> Solved in #256

在最近更新的版本 ( 2021/9/22 )，该问题仍然出现（掉落物品不显示），贴一份战斗结束后的cmd输出
```
 Ξ 离开结算画面
received message
sending notify wait {'duration': 1.0399353796038224, 'allow_skip': False}
received message
update value
sending notify wait {'duration': 0, 'allow_skip': False}
received message
update value
 Ξ
Traceback (most recent call last):
  File "E:\GoogleDown\ArknightsAutoHelper\Arknights\helper.py", line 463, in on_end_operation
    drops = imgreco.end_operation.recognize(smobj.prepare_reco['style'], screenshot, True)
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\end_operation.py", line 206, in recognize
    return recognize_main(im, learn_unrecognized_item)
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\end_operation.py", line 288, in recognize_main
    groupresult = tell_group(group, session, linetop, linebottom)
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\end_operation.py", line 72, in tell_group
    recognized_item = item.tell_item(itemimg, with_quantity=True, learn_unrecognized=session.learn_unrecognized)
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 171, in tell_item
    prob, item_id, name, item_type = get_item_id(common.convert_to_cv(itemimg))
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 86, in get_item_id
    ark_material_net, idx2id, id2idx, idx2name, idx2type = _load_net()
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 25, in _load_net
    idx2id, id2idx, idx2name, idx2type = load_index_info()
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 34, in load_index_info
    update_net()
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 59, in update_net
    resp = retry_get('https://cdn.jsdelivr.net/gh/triwinds/arknights-ml@latest/inventory/index_itemid_relation.json')
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 49, in retry_get
    raise ex
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 45, in retry_get
    return requests.get(url, timeout=timeout)
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\api.py", line 76, in get
    return request('get', url, params=params, **kwargs)
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\adapters.py", line 439, in send
    resp = conn.urlopen(
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\connectionpool.py", line 696, in urlopen
    self._prepare_proxy(conn)
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\connectionpool.py", line 964, in _prepare_proxy
    conn.connect()
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\connection.py", line 359, in connect
    conn = self._connect_tls_proxy(hostname, conn)
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\connection.py", line 500, in _connect_tls_proxy
    return ssl_wrap_socket(
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\util\ssl_.py", line 453, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls)
  File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\util\ssl_.py", line 495, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock)
  File "D:\Python\Python38\lib\ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "D:\Python\Python38\lib\ssl.py", line 997, in _create
    raise ValueError("check_hostname requires server_hostname")
ValueError: check_hostname requires server_hostname
 Ξ 第 8 次作战完成
received message
sending notify completed-count 8
sending notify wait {'duration': 9.209809597990002, 'allow_skip': True}
received message
received message
update value
received message
update value
```

**Comment by ZHDreamer** [comment]
我之前也这样，后来发现是因为开了代理，访问的网址timeout了
``` 
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 59, in update_net
     resp = retry_get('https://cdn.jsdelivr.net/gh/triwinds/arknights-ml@latest/inventory/index_itemid_relation.json')
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 49, in retry_get
     raise ex
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 45, in retry_get
     return requests.get(url, timeout=timeout)
```
把这个网址变成直连的就好了

**Comment by ZHDreamer** [comment]
> > Solved in #256
> 
> 在最近更新的版本 ( 2021/9/22 )，该问题仍然出现（掉落物品不显示），贴一份战斗结束后的cmd输出
> 
> ```
>  Ξ 离开结算画面
> received message
> sending notify wait {'duration': 1.0399353796038224, 'allow_skip': False}
> received message
> update value
> sending notify wait {'duration': 0, 'allow_skip': False}
> received message
> update value
>  Ξ
> Traceback (most recent call last):
>   File "E:\GoogleDown\ArknightsAutoHelper\Arknights\helper.py", line 463, in on_end_operation
>     drops = imgreco.end_operation.recognize(smobj.prepare_reco['style'], screenshot, True)
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\end_operation.py", line 206, in recognize
>     return recognize_main(im, learn_unrecognized_item)
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\end_operation.py", line 288, in recognize_main
>     groupresult = tell_group(group, session, linetop, linebottom)
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\end_operation.py", line 72, in tell_group
>     recognized_item = item.tell_item(itemimg, with_quantity=True, learn_unrecognized=session.learn_unrecognized)
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 171, in tell_item
>     prob, item_id, name, item_type = get_item_id(common.convert_to_cv(itemimg))
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 86, in get_item_id
>     ark_material_net, idx2id, id2idx, idx2name, idx2type = _load_net()
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 25, in _load_net
>     idx2id, id2idx, idx2name, idx2type = load_index_info()
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 34, in load_index_info
>     update_net()
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 59, in update_net
>     resp = retry_get('https://cdn.jsdelivr.net/gh/triwinds/arknights-ml@latest/inventory/index_itemid_relation.json')
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 49, in retry_get
>     raise ex
>   File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 45, in retry_get
>     return requests.get(url, timeout=timeout)
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\api.py", line 76, in get
>     return request('get', url, params=params, **kwargs)
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\api.py", line 61, in request
>     return session.request(method=method, url=url, **kwargs)
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\sessions.py", line 542, in request
>     resp = self.send(prep, **send_kwargs)
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\sessions.py", line 655, in send
>     r = adapter.send(request, **kwargs)
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\requests\adapters.py", line 439, in send
>     resp = conn.urlopen(
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\connectionpool.py", line 696, in urlopen
>     self._prepare_proxy(conn)
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\connectionpool.py", line 964, in _prepare_proxy
>     conn.connect()
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\connection.py", line 359, in connect
>     conn = self._connect_tls_proxy(hostname, conn)
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\connection.py", line 500, in _connect_tls_proxy
>     return ssl_wrap_socket(
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\util\ssl_.py", line 453, in ssl_wrap_socket
>     ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls)
>   File "E:\GoogleDown\ArknightsAutoHelper\venv\lib\site-packages\urllib3\util\ssl_.py", line 495, in _ssl_wrap_socket_impl
>     return ssl_context.wrap_socket(sock)
>   File "D:\Python\Python38\lib\ssl.py", line 500, in wrap_socket
>     return self.sslsocket_class._create(
>   File "D:\Python\Python38\lib\ssl.py", line 997, in _create
>     raise ValueError("check_hostname requires server_hostname")
> ValueError: check_hostname requires server_hostname
>  Ξ 第 8 次作战完成
> received message
> sending notify completed-count 8
> sending notify wait {'duration': 9.209809597990002, 'allow_skip': True}
> received message
> received message
> update value
> received message
> update value
> ```
我之前也这样，后来发现是因为开了代理，访问的网址timeout了
``` 
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 59, in update_net
     resp = retry_get('https://cdn.jsdelivr.net/gh/triwinds/arknights-ml@latest/inventory/index_itemid_relation.json')
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 49, in retry_get
     raise ex
  File "E:\GoogleDown\ArknightsAutoHelper\imgreco\item.py", line 45, in retry_get
     return requests.get(url, timeout=timeout)
```
把这个网址变成直连的就好了
